src/rust/catnap/mod.rs

```diff
@@ -458,12 +461,14 @@ impl CatnapLibOS {
         match qtable.get_mut(&qd) {
             Some(queue) => match queue.get_fd() {
                 Some(fd) => {
-                    // Create a closed socket. Should set this to a closing state.
-                    // FIXME: https://github.com/demikernel/demikernel/issues/625
-                    let closed_socket: Socket = {
-                        let socket: &Socket = queue.get_socket();
-                        socket.close()?
-                    };
+                    // Set socket as closing.
+                    let socket: &Socket = queue.get_socket();
+                    let closing_socket: Socket = socket.close()?;
+                    queue.set_socket(&closing_socket);
+
+                    // Create a closed socket.
```

Comment:
I think it would make more sense to put this into the async block since we already have the mutable queue reference there. Having it here is quite confusing, so I would recommend moving to line 488.

Reply:
Sure.